sound/soc/codecs/rt700-sdw.c

```diff
@@ -517,6 +535,23 @@ static int __maybe_unused rt700_dev_resume(struct device *dev)
 	regcache_sync_region(rt700->regmap, 0x3000, 0x8fff);
 	regcache_sync_region(rt700->regmap, 0x752010, 0x75206b);
 
+	if (!slave->unattach_request && rt700->hs_jack) {
+		regmap_write(rt700->regmap,
+			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D0);
+
+		regmap_write(rt700->regmap,
+			RT700_SET_MIC2_UNSOLICITED_ENABLE, 0x82);
+		regmap_write(rt700->regmap,
+			RT700_SET_HP_UNSOLICITED_ENABLE, 0x81);
+		regmap_write(rt700->regmap,
+			RT700_SET_INLINE_UNSOLICITED_ENABLE, 0x83);
+
+		dev_dbg(dev, "in %s jd enable\n", __func__);
+
+		regmap_write(rt700->regmap,
+			RT700_SET_AUDIO_POWER_STATE, AC_PWRST_D3);
```

Comment:
I guess this is correct somehow and has been tested, but just seems a bit odd... Is it really switching to D3 during resume?

Reply:
It is just the power-saving mode for the codec. The codec driver should go to power-saving mode before going to the suspend state. So, I just set back to power-saving mode during the resume function.

Reply:
ok, but we're resuming for playback, the codec goes to D3 here and then presumably it's switched back to D0 immediately, right? Is that common or has some specific reason? Naively I'd expect `.resume()` to bring the system to a fully operational mode to be able to begin operation immediately. Although bringing the hardware to some intermediate state, waiting for the actual operation to begin, should work too...